### ISSUE TITLE
Append parallel trees trained separately to existing model

### ISSUE THREAD

**Issue by ZiyueXu77** [dmlc/xgboost#8121]
I wonder if there is a mechanism to "append" several trees in a parallel/bagging manner to an existing model:
- The starting point will be a boosted bagging (num_parallel_trees = k > 0) model
- New k trees will be boosted in parallel and individually on k (fixed) data subsets based on the prediction of the starting model following this: https://xgboost.readthedocs.io/en/latest/python/examples/boost_from_prediction.html
- New k trees shall be appended to the existing model, but as parallel trees in 1 round

I currently did this by manually modifying the model json file, but the results does not appear to be correct. So I wonder if it is possible to achieve this, and how it may be done.

**Comment by trivialfis** [comment]
Unfortunately no, the gradient booster is a model as a whole and we don't expose the underlying base model(tree) to user interface. Could you please share the motivation of doing so? Why do you want to train trees with fixed dataset instead of sampled dataset.

**Comment by ZiyueXu77** [comment]
I see, thanks for the info!
I am trying to simulate a federated learning scenario where client will only share the boosted tree (instead of feature stats/gradients). So each client will boost their own tree based on local data - "fixed subset". And bagging seems to be the most direct way of doing "server model aggregation" under this setting.

**Comment by trivialfis** [comment]
> share the boosted tree

The tree splits on actual feature values. Wouldn't it be a major leak of data?

**Comment by trivialfis** [comment]
Have fun with the hack, let us know if there's anything we can help.
